aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -49,11 +53,10 @@ Tensor flip_cpu(const Tensor& self, IntList dims) {
         permute_order.emplace_back(i);
       }
     }
-    auto out_tensor = self.index(TensorList(final_indices));
-    return out_tensor.permute(IntList(permute_order));
+    std::sort(out_sizes_idx.begin(), out_sizes_idx.end(),
```

Comment:
There must be a better way to construct the order without doing an `O(n^2) + O(nlogn)` algorithm...